samples/snippets/src/main/java/com/example/bigquerystorage/WriteCommittedStream.java

```diff
@@ -45,54 +57,104 @@ public static void runWriteCommittedStream()
+      System.out.println("Failed to append records. \n" + e);
+    }
+
+    // Final cleanup.
+    writer.cleanup();
+    System.out.println("Appended records successfully.");
+  }
+
+  public static WriteCommittedStream create(String projectId, String datasetName, String tableName)
+      throws IOException, DescriptorValidationException, InterruptedException {
+    WriteCommittedStream writer = new WriteCommittedStream();
+    writer.client = BigQueryWriteClient.create();
+    // Initialize a write stream for the specified table.
+    // For more information on WriteStream.Type, see:
+    // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1beta2/WriteStream.Type.html
+    WriteStream stream = WriteStream.newBuilder().setType(WriteStream.Type.COMMITTED).build();
+    TableName parentTable = TableName.of(projectId, datasetName, tableName);
+    CreateWriteStreamRequest createWriteStreamRequest =
+        CreateWriteStreamRequest.newBuilder()
+            .setParent(parentTable.toString())
+            .setWriteStream(stream)
+            .build();
+    writer.writeStream = writer.client.createWriteStream(createWriteStreamRequest);
+
+    // Use the JSON stream writer to send records in JSON format.
+    // For more information about JsonStreamWriter, see:
+    // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.html
+    writer.streamWriter =
+        JsonStreamWriter.newBuilder(
+                writer.writeStream.getName(), writer.writeStream.getTableSchema())
+            .build();
+    return writer;
+  }
+
+  public void append(JSONArray data, int offset)
+      throws DescriptorValidationException, IOException, ExecutionException, InterruptedException {
+    // To detect duplicate records, pass the index as the record offset.
+    // To disable deduplication, omit the offset or use WriteStream.Type.DEFAULT.
+    ApiFuture<AppendRowsResponse> future = streamWriter.append(data, /*offset=*/ offset);
+    this.inflightRequestPhaser.register();
+    ApiFutures.addCallback(
+        future, new AppendCompleteCallback(this), MoreExecutors.directExecutor());
+  }
+
+  public void cleanup() {
+    // Wait for all in-flight requests to complete.
+    this.inflightRequestPhaser.arriveAndAwaitAdvance();
+    // Finalize the stream after use.
+    FinalizeWriteStreamRequest finalizeWriteStreamRequest =
+        FinalizeWriteStreamRequest.newBuilder().setName(writeStream.getName()).build();
+    client.finalizeWriteStream(finalizeWriteStreamRequest);
+  }
+
+  class AppendCompleteCallback implements ApiFutureCallback<AppendRowsResponse> {
+
+    private final WriteCommittedStream parent;
+
+    AppendCompleteCallback(WriteCommittedStream parent) {
+      this.parent = parent;
+    }
+
+    public void onSuccess(AppendRowsResponse response) {
+      if (response.hasError()) {
```

Comment:
I don't know what to do here...Maybe add a comment of how this can be retried? If we don't have a good retry case, should we have this sample?

Reply:
My $0.02: I think it's better to show how to check the AppendRowsResponse error status, rather than not show anything here.

Reply:
IIUC StreamWriter already handles this field which will invoke onFailure instead 
https://github.com/googleapis/java-bigquerystorage/blob/54874be945c2b88be6be03ae654277445c17741d/google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java#L565

I updated that code to show a little error handling; not sure if there's something more specific to show there